File: src/main/java/MoveZeroes/OnePunchMan34/MoveZeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        #postion of output
        pos=0
        for i in range(len(nums)):
            if nums[i]:
                nums[pos]=nums[i]
                pos+=1
        for i in range(pos,len(nums)):
            nums[i]=0

File: src/main/java/MoveZeroes/wuydit/MoveZeroes.java
package MoveZeroes.wuydit;

public class MoveZeroes {
    public void moveZeroes(int[] nums) {
        int zero = -1;
        int next = 0;
        while (next < nums.length) {
            if (nums[next] == 0) {
                if (zero == -1) {
                    zero = next;
                }
            } else {
                if (zero >= 0) {
                    nums[zero] = nums[next];
                    nums[next] = 0;
                    zero++;
                }
            }
            next++;
        }
    }
}
